src/kiara_plugin/topic_modelling/modules/corpus_preparation.py

```diff
@@ -30,83 +25,221 @@ def create_inputs_schema(self):
+                    ]
                 )
-        except Exception as e:
-            print(e)
-            pass
+
+            except Exception as e:
+                msg = f"An error occurred while augmenting the dataframe: {e}"
+                raise KiaraProcessingException(msg)
 
         try:
-            output_table = sources.to_arrow()
+            output_table = augm_sources.to_arrow()
 
         except Exception as e:
-            print(f"An error occurred: {e}")
+            raise KiaraProcessingException(e)
+
+        outputs.set_value("corpus_table", output_table)
+
+
+class CorpusDistTime(KiaraModule):
+    """
+    This module aggregates a table by day, month or year from a corpus table that contains a date column. It returns the distribution over time, which can be used for display purposes, such as visualization.
+    
+    """
+
+    _module_type_name = "topic_modelling.time_dist"
 
-        outputs.set_value("corpus_table", output_table)
\ No newline at end of file
+    def create_inputs_schema(self):
+
+        return {
+            "periodicity": {
+                "type": "string",
+                "type_config": {"allowed_strings": ["day", "month", "year"]},
+                "doc": "The desired data periodicity to aggregate the data. Values can be either 'day','month' or 'year'.",
+            },
+            "date_col_name": {
+                "type": "string",
+                "doc": "Column name of the column that contains the date. Values in this column need to comply with date format: https://docs.rs/chrono/latest/chrono/format/strftime/index.html.",
+            },
+            "title_col_name": {
+                "type": "string",
+                "doc": "Column name of the values containing publication names or ref/id. This column will be used in the output.",
+            },
+            "corpus_table": {
+                "type": "table",
+                "doc": "The corpus table for which the distribution over time is needed.",
+            },
+        }
+
+    def create_outputs_schema(self):
+        return {"dist_table": {"type": "table", "doc": "The aggregated data table."}}
+
+    def process(self, inputs, outputs) -> None:
+        
+        import duckdb
+        import polars as pl
+        import pyarrow as pa
+
+        agg = inputs.get_value_obj("periodicity").data
+        title_col = inputs.get_value_obj("title_col_name").data
+        time_col = inputs.get_value_obj("date_col_name").data
+        table_obj = inputs.get_value_obj("corpus_table")
+
+        if not agg:
+            raise KiaraProcessingException("No periodicity provided.")
+        
+        if not title_col:
+            raise KiaraProcessingException("No column name provided for titles id/ref.")
+        
+        if not time_col:
+            raise KiaraProcessingException("No column name provided for date.")
+        
+        if not table_obj:
```

Comment:
Same as before, you only test the 'Value' instance, not its data, so it will never be `False`. For all the others you test the `data` property.